lib/apiservers/engine/backends/container_proxy.go

```diff
@@ -317,6 +324,104 @@ func (c *ContainerProxy) CreateExecTask(handle string, config *types.ExecConfig)
 	return handleprime, eid, nil
 }
 
+func (c *ContainerProxy) InspectTask(op trace.Operation, handle string, eid string, cid string) (*models.TaskInspectResponse, error) {
+	defer trace.End(trace.Begin(fmt.Sprintf("handle(%s), eid(%s), cid(%s)", handle, eid, cid)))
+
+	// inspect the Task to obtain ProcessConfig
+	config := &models.TaskInspectConfig{
+		Handle: handle,
+		ID:     eid,
+	}
+
+	// FIXME: right now we are only using this path for exec targets. But later the error messages may need to be changed
+	// to be more accurate.
+	params := tasks.NewInspectParamsWithContext(ctx).WithConfig(config)
+	resp, err := c.client.Tasks.Inspect(params)
+	if err != nil {
+		switch err := err.(type) {
+		case *tasks.InspectNotFound:
+			// These error types may need to be expanded. NotFoundError does not fit here.
+			op.Errorf("received a TaskNotFound error during task inspect: %s", err.Payload.Message)
+			return nil, ConflictError(fmt.Sprintf("container (%s) has been poweredoff", cid))
```

Comment:
I'm not sure turning this into a conflict error is the right thing to do in the proxy. While it could be used to force a handle refresh it could also lead to looping behaviour if the task remains not found.

If we are to return a conflict error from here I think it should be the port layer that generates it - and does so by checking the handle provided against it's cache state. If there's a conflict in changeVersion then a conflict error is legitimate. If there isn't then it's just a mistake in the eid/cid that's used.

Reply:
Agreed, I will likely look at adding a different error type that is translated to a valid error in `container.go` since these error messages are adhering to the docker client return codes.